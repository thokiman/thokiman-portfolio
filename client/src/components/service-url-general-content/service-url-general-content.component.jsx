import React from "react";

import {
  ServiceDescription,
  ServiceDescriptionContainer,
  ServiceDescriptionIconContainer,
  ServiceDescriptionIconItem,
  ServiceDescriptionItem,
  ServiceDescriptionTitle,
  ServiceGeneralContent,
  ServiceTitle,
} from "./service-url-general-content.styles";

export const ServiceUrlGeneralContent = ({ title, description }) => {
  return (
    <ServiceGeneralContent>
      <ServiceTitle>{title}</ServiceTitle>
      <ServiceDescription>
        {description.map(({ id, title, description, items }) => {
          return (
            <ServiceDescriptionContainer key={id}>
              <ServiceDescriptionTitle>{title}</ServiceDescriptionTitle>
              <ServiceDescriptionItem>{description}</ServiceDescriptionItem>
              <ServiceDescriptionIconContainer>
                {items.map(({ id, name, iconPath }) => {
                  return (
                    <ServiceDescriptionIconItem
                      key={id}
                      src={iconPath}
                      alt={name}
                    />
                  );
                })}
              </ServiceDescriptionIconContainer>
            </ServiceDescriptionContainer>
          );
        })}
      </ServiceDescription>
    </ServiceGeneralContent>
  );
};

export default ServiceUrlGeneralContent;
