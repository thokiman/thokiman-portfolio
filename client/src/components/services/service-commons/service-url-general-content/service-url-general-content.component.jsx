import React from 'react';
import {
  ServiceDescription,
  ServiceDescriptionContainer,
  ServiceDescriptionItem,
  ServiceDescriptionTitle,
  ServiceTitle,
  ServiceUrlGeneralContainer,
} from './service-url-general-content.styles';

export const ServiceUrlGeneralContent = ({ title, description }) => {
  return (
    <ServiceUrlGeneralContainer>
      <ServiceTitle className='service-url-general-content-title'>
        {title}
      </ServiceTitle>
      <ServiceDescription>
        {description.map(({ id, title, description }) => {
          return (
            <ServiceDescriptionContainer key={id}>
              <ServiceDescriptionTitle className='service-url-general-content-description'>
                {title}
              </ServiceDescriptionTitle>
              <ServiceDescriptionItem className='service-url-general-content-description'>
                {description}
              </ServiceDescriptionItem>
            </ServiceDescriptionContainer>
          );
        })}
      </ServiceDescription>
    </ServiceUrlGeneralContainer>
  );
};

export default ServiceUrlGeneralContent;
