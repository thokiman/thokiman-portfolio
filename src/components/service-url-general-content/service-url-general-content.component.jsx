import React from "react";

import "./service-url-general-content.styles.scss";

const ServiceUrlGeneralContent = ({ title, description }) => {
  return (
    <div className="service-general-content">
      <div className="service-title">{title}</div>
      <div className="service-description">
        {description.map(({ id, title, description, items }) => {
          return (
            <div key={id} className="service-description-container">
              <div className="service-description-title">{title}</div>
              <div className="service-description-item">{description}</div>
              <div className="service-description-icon-container">
                {items.map(({ id, name, iconPath }) => {
                  return (
                    <img
                      className="service-description-icon-item"
                      key={id}
                      src={iconPath}
                      alt={name}
                    />
                  );
                })}
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default ServiceUrlGeneralContent;
