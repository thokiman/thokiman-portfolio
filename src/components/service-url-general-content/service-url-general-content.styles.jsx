import styled, { keyframes } from "styled-components";

export const ServiceGeneralContent = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
`;

const serviceTitle = keyframes`
0% {
                 color:  rgba(116,198,212,0.8);
             }

             100% {
                 color: rgba(227,225,228,1.0);
             }

`;
export const ServiceTitle = styled.div`
  font-family: futura-demi;
  font-size: 8vw;
  margin-bottom: 20px;
  width: 100%;
  height: 100%;
  color: #e3e1e4;
  letter-spacing: 0.1em;
  font-kerning: auto;

  animation: ${serviceTitle} 3s linear infinite alternate both;
`;

export const ServiceDescription = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
  margin-bottom: 20px;
  width: 94%;
  height: 100%;
`;

export const ServiceDescriptionContainer = styled.div`
  margin-bottom: 15px;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const ServiceDescriptionTitle = styled.div`
  font-family: futura-demi;
  font-size: 4vw;
  margin-bottom: 15px;
  text-indent: 0.6em;
  color: rgba(188, 207, 213, 0.9);
  letter-spacing: 0.02em;
  font-kerning: auto;
`;
export const ServiceDescriptionItem = styled.div`
  font-size: 2.5vw;
  text-align: left;
  color: rgba(227, 225, 228, 0.9);
  letter-spacing: 0.06em;
  line-height: 1.5em;
  font-kerning: auto;
  margin-bottom: 30px;
`;
export const ServiceDescriptionIconContainer = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 30px;
  border-bottom: #74c6d4 solid 1px;
`;
export const ServiceDescriptionIconItem = styled.img`
  width: 27%;
  height: 27%;
`;

ServiceGeneralContent.displayName = "ServiceGeneralContent";
ServiceTitle.displayName = "ServiceTitle";
ServiceDescription.displayName = "ServiceDescription";
ServiceDescriptionContainer.displayName = "ServiceDescriptionContainer";
ServiceDescriptionTitle.displayName = "ServiceDescriptionTitle";
ServiceDescriptionItem.displayName = "ServiceDescriptionItem";
ServiceDescriptionIconContainer.displayName = "ServiceDescriptionIconContainer";
ServiceDescriptionIconItem.displayName = "ServiceDescriptionIconItem";
