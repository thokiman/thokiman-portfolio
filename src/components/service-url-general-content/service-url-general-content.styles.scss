 .service-general-content {
     display: flex;
     flex-direction: column;
     justify-content: center;
     align-items: flex-start;


     .service-title {
         font-family: futura-demi;
         font-size: 430%;
         margin-bottom: 20px;
         width: 100%;
         height: 100%;
         color: #e3e1e4;
         letter-spacing: 0.1em;
         font-kerning: auto;

         animation: color-change-2x 3s linear infinite alternate both;

         @keyframes color-change-2x {
             0% {
                 color: rgba($color: #74c6d4, $alpha: 0.8);
             }

             100% {
                 color: rgba($color:#e3e1e4, $alpha: 1);
             }
         }
     }

     .service-description {
         display: flex;
         flex-direction: column;
         align-items: center;
         justify-content: center;
         margin: 0 auto;
         margin-bottom: 20px;
         width: 94%;
         height: 100%;

         .service-description-container {
             margin-bottom: 15px;
             display: flex;
             flex-direction: column;
             justify-content: center;

             .service-description-title {
                 font-family: futura-demi;
                 font-size: 240%;
                 text-indent: 0.6em;
                 color: rgba($color:#bccfd5, $alpha:0.9);
                 letter-spacing: 0.02em;
                 font-kerning: auto;
             }

             .service-description-item {
                 font-size: 150%;
                 text-align: justify;
                 color: rgba($color:#e3e1e4, $alpha:0.9);
                 letter-spacing: 0.06em;
                 line-height: 1.5em;
                 font-kerning: auto;
                 margin-bottom: 30px;
             }

             .service-description-icon-container {
                 display: flex;
                 justify-content: space-between;
                 padding: 30px;
                 border-bottom: #74c6d4 solid 1px;

                 .service-description-icon-item {
                     width: 27%;
                     height: 27%;
                 }
             }
         }
     }
 }