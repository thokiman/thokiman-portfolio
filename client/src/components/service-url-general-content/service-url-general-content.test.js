import React from "react";
import { shallow } from "enzyme";

import { ServiceUrlGeneralContent } from "./service-url-general-content.component";

describe("ServiceUrlGeneralContent component", () => {
  let wrapper;
  const mockArray = [];
  const mockText = "";

  beforeEach(() => {
    const mockProps = {
      title: mockText,
      description: mockArray,
    };
    wrapper = shallow(<ServiceUrlGeneralContent {...mockProps} />);
  });

  it("should render ServiceUrlGeneralContent component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
