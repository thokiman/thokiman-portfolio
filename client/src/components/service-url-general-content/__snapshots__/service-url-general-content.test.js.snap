// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceUrlGeneralContent component should render ServiceUrlGeneralContent component 1`] = `ShallowWrapper {}`;
